tests/integration/API/WorkController.test.ts

```diff
@@ -0,0 +1,488 @@
+import { delay, runtimeId, createDatabase } from '../../helpers/utils'
+import { getWork, postWork } from '../../helpers/works'
+
+const PREFIX = `test-functional-node-poet-${runtimeId()}`
+const NODE_PORT = '28081'
+const getWorkFromNode = getWork(NODE_PORT)
+const postWorkToNode = postWork(NODE_PORT)
+
+const works = [
+  ABraveAndStartlingTruth,
+  AllGodsChildrenNeedTravelingShoes,
+  ASongFlungUpToHeaven,
+  AStudyInScarlet,
+  GatherTogetherInMyName,
+  KnowWhyTheCagedBirdSings,
+  MomAndMeAndMom,
+  OnThePulseOfMorning,
+  SinginAndSwinginAndGettingMerryLikeChristmas,
+  TheHeartOfAWoman,
+  TheMurdersInTheRueMorgue,
+  TheWeekOfDiana,
+]
+
+const blockchainSettings = {
+  MINIMUM_BLOCK_HEIGHT: 100,
+  ENABLE_TIMESTAMPING: true,
+  TIMESTAMP_INTERVAL_IN_SECONDS: 10,
+  BATCH_CREATION_INTERVAL_IN_SECONDS: 5,
+  READ_DIRECTORY_INTERVAL_IN_SECONDS: 5,
+  UPLOAD_CLAIM_INTERVAL_IN_SECONDS: 5,
+}
+
+const bitcoindClient = new Client({
+  host: process.env.BITCOIN_URL || 'bitcoind-1',
+  port: 18443,
+  network: 'regtest',
+  password: 'bitcoinrpcpassword',
+  username: 'bitcoinrpcuser',
+})
+
+const setUpServerAndDb = async () => {
+  const db = await createDatabase(PREFIX)
+  const server = await app({
+    BITCOIN_URL: process.env.BITCOIN_URL || 'bitcoind-1',
+    API_PORT: NODE_PORT,
+    MONGODB_DATABASE: db.settings.tempDbName,
+    MONGODB_USER: db.settings.tempDbUser,
+    MONGODB_PASSWORD: db.settings.tempDbPassword,
+    EXCHANGE_PREFIX: PREFIX,
+    ...blockchainSettings,
+  })
+
+  return { db, server }
+}
+
+const getClaimWithoutTimestamp = (claim: any) => pickBy((v: any, k: string) => k !== 'timestamp', claim)
+const getClaimsWithoutTimestamps = (claims: ReadonlyArray<any>) => claims.map(getClaimWithoutTimestamp)
+
+const setUpExistingClaims = async (claims: ReadonlyArray<Claim>) =>
+  await Promise.all(claims.map(async (claim: Claim) => postWorkToNode(claim)))
+
+describe('GET /works/:id', async assert => {
+  // Setup Mongodb and the app server
+  const { db, server } = await setUpServerAndDb()
+
+  // Make sure node has regtest coins to pay for transactions.
+  await ensureBitcoinBalance(bitcoindClient)
+
+  // Allow everything to finish starting.
+  await delay(5 * 1000)
+
+  // Submit a claim
+  await setUpExistingClaims([TheRaven])
+
+  // Wait for a claim batch to be submitted to the blockchain.
+  await delay(parseInt(process.env.TIMESTAMP_INTERVAL_IN_SECONDS || '10', 10) * 1000 * 2)
+
+  // mine N confirmation blocks on bitcoindA.
+  await bitcoindClient.generate(parseInt(process.env.CONFIRMATION_BLOCKS || '1', 10))
+
+  // Wait for claim batches to be read from the blockchain.
+  await delay(parseInt(process.env.READ_DIRECTORY_INTERVAL_IN_SECONDS || '5', 10) * 1000 * 2)
+
+  {
+    let theRavenClaim
+    let theRavenResponse
+    try {
+      theRavenResponse = await getWorkFromNode(TheRaven.id)
+      theRavenClaim = await theRavenResponse.json()
+    } catch (err) {
+      theRavenClaim = {}
```

Comment:
Would this provide a better failure report if you set this to the error so you can see it in the `actual` vs `expected` report?